tools/etcdhelper/etcdhelper.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"os"
+
+	etcd "github.com/coreos/etcd/clientv3"
+	"golang.org/x/net/context"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime/serializer/json"
+	"k8s.io/kubernetes/pkg/runtime/serializer/protobuf"
+
+	// install all APIs
+	_ "github.com/openshift/origin/pkg/api/install"
+	_ "github.com/openshift/origin/pkg/quota/api/install"
+	_ "k8s.io/kubernetes/pkg/api/install"
+)
+
+func main() {
+	var endpoint, keyFilePath, certFilePath string
+	flag.StringVar(&endpoint, "endpoint", "https://127.0.0.1:4001", "Etcd endpoint.")
+	flag.StringVar(&keyFilePath, "key-file", "", "TLS client key.")
+	flag.StringVar(&certFilePath, "cert-file", "", "TLS client certitificate.")
+	flag.Parse()
+
+	if flag.NArg() == 0 {
+		fmt.Fprintf(os.Stderr, "ERROR: you need to specify action ls [<key>] or get <key>\n")
+		os.Exit(1)
+	}
+	if flag.Arg(0) == "get" && flag.NArg() == 1 {
+		fmt.Fprintf(os.Stderr, "ERROR: you need to specify <key> for get operation\n")
+		os.Exit(1)
+	}
+	action := flag.Arg(0)
+	key := ""
+	if flag.NArg() > 1 {
+		key = flag.Arg(1)
+	}
+
+	cert, err := tls.LoadX509KeyPair(certFilePath, keyFilePath)
```

Comment:
If you don't pass it, you won't be able to connect to it (assuming you're running default openshift version), although I don't see any problems making them optional.